boot/freeldr/freeldr/ntldr/winldr.c

```diff
@@ -1251,6 +1251,29 @@ LoadAndBootWindowsCommon(
     /* "Stop all motors", change videomode */
     MachPrepareForReactOS();
 
+    /* Show the "debug mode" notice if needed */
+    BOOLEAN DebugMode = !NtLdrGetOption(BootOptions, "NODEBUG") && /* NODEBUG is absent, priority over DEBUG option */
+                        !NtLdrGetOption(BootOptions, "CRASHDEBUG") && /* CRASHDEBUG is abset, no wait during the boot */
+                        !!NtLdrGetOption(BootOptions, "DEBUG"); /* DEBUG is present */
+    if (DebugMode)
+    {
+        /* It is booting in debug mode */
+        TuiPrintf("Booting in debug mode\n");
+
+        /* Check whether there is a DEBUGPORT option */
+        PCSTR DebugPort;
+        ULONG DebugPortLength;
+        DebugPort = NtLdrGetOptionEx(BootOptions, "DEBUGPORT=", &DebugPortLength);
+        if (DebugPort && (DebugPortLength > 10))
+        {
+            /* Show the debug port */
+            DebugPort += 10;
+            DebugPortLength -= 10;
+            TuiPrintf("You would need a debugger attached to the %.*s port to continue.\n", DebugPortLength, DebugPort);
```

Comment:
I guess some kind of message is wanted when there is no explicit port too...

Reply:
If no port is provided by user, it would default to "COM", so if it's actually going to boot in debug mode at least "DEBUGPORT=COM" is present. If it's not present it's unlikely that kernel would be waiting for debugger connection.

Here FreeLoader inserts `DEBUGPORT=COM` when no DEBUGPORT is provided: https://github.com/reactos/reactos/blob/d06d9f3e8f63e5a4b610e9605b40da20990e23a2/boot/freeldr/freeldr/ntldr/winldr.c#L841

Reply:
"inserts" it doesn't insert anything, it just sets these two local parameters to these values so as to then generate the default "kdcom.dll" name. But the boot options line is left verbatim.

Reply:
My bad, I misunderstood its purpose.